rust/worker/src/execution/operators/brute_force_knn.rs

```diff
@@ -106,22 +110,23 @@ impl Operator<BruteForceKnnOperatorInput, BruteForceKnnOperatorOutput> for Brute
             let log_record = data.0;
             let index = data.1;
 
-            if log_record.record.operation == Operation::Delete {
+            if log_record.final_operation == Operation::Delete {
                 // Explicitly skip deleted records.
                 continue;
             }
 
             // Skip records that are disallowed. If allowed list is empty then
             // don't exclude anything.
             // Empty allowed list is passed when where filtering is absent.
+            // TODO: This should not need to use merged_user_id, which clones the id.
             if !input.allowed_ids.is_empty()
                 && !input
                     .allowed_ids_brute_force
-                    .contains(&log_record.record.id)
+                    .contains(&log_record.merged_user_id())
```

Comment:
There is a `merged_user_id_ref()` that does not clone() that can be used here probably?

Reply:
Not without changing what we are calling contains() on, that is Arc<[String]> so it can't take str without clone anyway

Reply:
I've added a TODO to restructure this a bit.